odoo/tools/js_transpiler.py

```diff
@@ -253,7 +253,8 @@ def repl(matchobj):
     ^
     (?P<space>\s*)                      # space and empty line
     export\s*                           # export
-    (?P<object>{(\s*\w+\s*,?\s*)*})\s*  # { a, b, c as x, ... }
+    (?P<object>{((\s*\w+\s*)|(\s*\w+\s+as\s+\w+\s*))
+        ((,\s*\w+\s*)|(,\s*\w+\s+as\s+\w+\s*))*})\s*  # { a, b, c as x, ... }
```

Comment:
That seems like a rather complicated way to work around the backtracking issue.

We don't seem to use the sub-matches inside `<object>`, is there really a point of making the regex so specific?

It could be much simpler to match a dumb "character, whitespace and commas soup" inside the curly braces: `[\w\s,]+`
There would be no backtracking problems as there's no nested repetitions anymore, and the regex would be shorter and easier to read. I don't think there is any risk of mismatch either, considering the `export {...} from` anchoring it.

A [quick test](https://regex101.com/r/1yw8aa/1) shows faster matching too (83 steps vs 439 steps for a 3 lines sample), but that's just a bonus.

```suggestion
    (?P<object>{[\w\s,]+})\s*  # { a, b, c as x, ... }
```

Note: the same optimization is possible for `EXPORT_OBJECT_RE` above.

Reply:
Thx for the idea :)